src/pages-and-resources/discussions/app-config-form/AppConfigForm.jsx

```diff
@@ -1,51 +1,58 @@
 import React, {
-  useCallback, useContext, useEffect, useRef,
+  useCallback, useContext, useEffect,
 } from 'react';
 import PropTypes from 'prop-types';
 import { useDispatch, useSelector } from 'react-redux';
 import { useRouteMatch } from 'react-router';
-import { Container } from '@edx/paragon';
 import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
 import { history } from '@edx/frontend-platform';
+import { Container } from '@edx/paragon';
+
 import { useModel } from '../../../generic/model-store';
-import { fetchAppConfig, saveAppConfig } from './data/thunks';
-import LegacyConfigForm from './apps/legacy';
-import LtiConfigForm from './apps/lti';
-import messages from './messages';
 import { PagesAndResourcesContext } from '../../PagesAndResourcesProvider';
-import AppConfigFormProvider from './AppConfigFormProvider';
+import { LOADED, selectApp } from '../data/slice';
+import { saveAppConfig } from '../data/thunks';
+
+import messages from './messages';
+import AppConfigFormProvider, { AppConfigFormContext } from './AppConfigFormProvider';
 import AppConfigFormApplyButton from './AppConfigFormApplyButton';
+import LegacyConfigForm from './apps/legacy';
+import LtiConfigForm from './apps/lti';
 
 function AppConfigForm({
   courseId, intl,
 }) {
-  const { params: { appId: routeAppId } } = useRouteMatch();
-  const formRef = useRef();
+  const dispatch = useDispatch();
+  const { formRef } = useContext(AppConfigFormContext);
   const { path: pagesAndResourcesPath } = useContext(PagesAndResourcesContext);
+  const { params: { appId: routeAppId } } = useRouteMatch();
+  const { selectedAppId, status } = useSelector(state => state.discussions);
+  const app = useModel('apps', selectedAppId);
+  // appConfigs have no ID of their own, so we use the active app ID to reference them.
+  // This appConfig may come back as null if the selectedAppId is not the activeAppId, i.e.,
+  // if we're configuring a new app.
+  const appConfig = useModel('appConfigs', selectedAppId);
 
-  const dispatch = useDispatch();
   useEffect(() => {
-    dispatch(fetchAppConfig(courseId, routeAppId));
-  }, [courseId]);
-
-  const { appId, appConfigId } = useSelector(state => state.discussions.appConfigForm);
+    if (status === LOADED) {
+      if (routeAppId !== selectedAppId) {
+        dispatch(selectApp({ appId: routeAppId }));
```

Comment:
This allows the config form to set the selectedAppId if it's loaded directly without the user having gone through step 1 (which is possible, since each step has its own route)